debian/autobake-deb.sh

```diff
@@ -50,40 +50,82 @@ remove_rocksdb_tools()
   fi
 }
 
-
-CODENAME="$(lsb_release -sc)"
-case "${CODENAME}" in
-	stretch)
-		# MDEV-28022 libzstd-dev-1.1.3 minimum version
-		sed -i -e '/libzstd-dev/d' \
-                       -e 's/libcurl4/libcurl3/g' -i debian/control
-		remove_rocksdb_tools
-		;;
-	bionic)
-		remove_rocksdb_tools
-		;;
-esac
-
-if [[ ! "$(dpkg-architecture -q DEB_BUILD_ARCH)" =~ amd64|arm64|ppc64el|s390x ]]
-then
-  remove_rocksdb_tools
-fi
-
-# From Debian Bullseye/Ubuntu Groovy, liburing replaces libaio
-if ! apt-cache madison liburing-dev | grep 'liburing-dev' >/dev/null 2>&1
-then
+replace_uring_with_aio()
+{
   sed 's/liburing-dev/libaio-dev/g' -i debian/control
-  sed '/-DIGNORE_AIO_CHECK=YES/d' -i debian/rules
-  sed '/-DWITH_URING=yes/d' -i debian/rules
-fi
+  sed -e '/-DIGNORE_AIO_CHECK=YES/d' \
+      -e '/-DWITH_URING=yes/d' -i debian/rules
+}
 
-# From Debian Buster/Ubuntu Focal onwards libpmem-dev is available
-# Don't reference it when built in distro releases that lack it
-if ! apt-cache madison libpmem-dev | grep 'libpmem-dev' >/dev/null 2>&1
-then
+disable_pmem()
+{
   sed '/libpmem-dev/d' -i debian/control
   sed '/-DWITH_PMEM=yes/d' -i debian/rules
-fi
+}
+
+architecture=$(dpkg-architecture -q DEB_BUILD_ARCH)
+
+CODENAME="$(lsb_release -sc)"
+case "${CODENAME}" in
+  stretch)
+    # MDEV-16525 libzstd-dev-1.1.3 minimum version
+    sed -e '/libzstd-dev/d' \
+        -e 's/libcurl4/libcurl3/g' -i debian/control
+    remove_rocksdb_tools
+    disable_pmem
+    ;&
+  buster)
+    replace_uring_with_aio
+    if [ ! "$architecture" = amd64 ]
+    then
+      disable_pmem
+    fi
+    ;&
+  bullseye|bookworm)
```

Comment:
For the time being, bookworm=sid and should just work out-of-the-box.

I still think using `apt-cache madison` was a good system and apt-cache to my knowledge always works if the build host is healthy.